python/ray/serve/kv_store.py

```diff
@@ -56,3 +60,143 @@ def delete(self, key):
         if not isinstance(key, str):
             raise TypeError("key must be a string, got: {}.".format(type(key)))
         return ray_kv._internal_kv_del(self._format_key(key))
+
+
+class RayExternalKVStore:
```

Comment:
We should add a common base class to define the interface for all of the impls here. You can use an abstract base class for this:
https://www.geeksforgeeks.org/abstract-classes-in-python/